examples/calculate_due_date.py

```diff
@@ -0,0 +1,95 @@
+import re
+import datetime
+import zkit
+from functools import cache
+from zscript import dataset
+
+
+def extract_number(text):
+    """Extracts the number from a text string."""
+    match = re.search(r'\d+', text)
+    if match:
+        return int(match.group())
+    return None
+
+
+def regex_search(string, pattern):
+    """Checks if the string contains 'Upon Receipt'"""
+    return re.search(pattern, string, re.IGNORECASE)
+
+
+def add_days_str(start_date_str, days):
+    """Adds a specified number of days to a given date string and returns the result as a string.
+
+    Args:
+      start_date_str: The starting date as a string.
+      days: The number of days to add.
+
+    Returns:
+      A string representing the date after adding the specified days in the specified format.
+    """
+
+    start_date = datetime.datetime.strptime(start_date_str, "%Y-%m-%d")
+    end_date = start_date + datetime.timedelta(days=days)
+    return end_date.strftime("%Y-%m-%d")
+
+
+def get_first_of_next_month_as_string(date_str: str):
+    """Gets the first day of the next month from a given date as a string.
+
+    Args:
+      date: The input date str.
+
+    Returns:
+      A string representing the first day of the next month in the format "YYYY-MM-DD".
+    """
+    dt = datetime.datetime.strptime(date_str, "%Y-%m-%d")
+    first_of_month_date = (dt.replace(day=1) + datetime.timedelta(days=32)).replace(day=1)
+    return first_of_month_date.strftime("%Y-%m-%d")
+
+
+@cache
+def get_doc_created_at_date(doc_id):
+    return zkit.get_obj(f'/v2/pd/documents/{doc_id}', query={'fields__only': 'id,created_at'})['created_at']
+
+# Calculates the due date if there isn't a due date already set.
```

Comment:
I am wondering if instead of a comment, we do a doc-comment on the function, like so:

```python
def calculate_due_date(value, field, needed_fields):
    """ Calculates the due date if there isn't a due date already set.
    """
```

(keep the comments inside the function, I am just talking about the comment that does th general description of the function)